iothub/device/src/DeviceClient.cs

```diff
@@ -707,10 +707,11 @@ protected virtual void Dispose(bool disposing)
         /// <summary>
         /// Retrieve the device properties.
         /// </summary>
+        /// <param name="payloadConvention">A convention handler that defines the content encoding and serializer to use for commands.</param>
         /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
         /// <returns>The device properties.</returns>
-        public Task<Properties> GetPropertiesAsync(CancellationToken cancellationToken)
-            => InternalClient.GetPropertiesAsync(cancellationToken);
+        public Task<Properties> GetPropertiesAsync(IPayloadConvention payloadConvention = default, CancellationToken cancellationToken = default)
```

Comment:
Should we reverse the order of the parameters here?

Below we have payload convention coming before the cancellation token, but it might be more natural to write
```c#
// With them reversed
GetPropertiesAsync();
GetPropertiesAsync(cts);
GetPropertiesAsync(cts, convention);

// With them as they are
GetPropertiesAsync(convention, cts);
GetPropertiesAsync(null, cts);
GetPropertiesAsync(cancellationToken: cts);
```

Reply:
In my mind I always put cancellation token at the end of any parameter list (baring an actual `params` being passed), so I like it the way we have it here. I'd be comfortable with `GetPropertiesAsync(cancellationToken: cts);`, but this is definitely a personal preference.